src/rospkg/os_detect.py

```diff
@@ -215,6 +215,24 @@ def get_codename(self):
         raise OsNotDetected("called in incorrect OS")
 
 
+class OpenEmbedded(OsDetector):
+    """
+    Detect OpenEmbedded.
+    """
+    def is_os(self):
+        return "ROS_OS_OVERRIDE" in os.environ and os.environ["ROS_OS_OVERRIDE"] == "oe"
```

Comment:
Although this will work for the current use case of needing to support rosdep during the cross compile stage. I suspect that we can actually detect OpenEmbedded systems at runtime which will enable more use cases down the line. For example rosdep could be able to validate dependencies are available even if it can't install them on a running system.

Poking around I found that ther e does look to be an /etc/issue generated: http://cgit.openembedded.org/openembedded-core/tree/meta/recipes-core/base-files/base-files_3.0.14.bb?h=master

Reply:
@tfoote << Many thanks for your review and insights!

Not sure if running rosdep on the target side makes sense _yet_.
As per your example, what would you do with this information? I guess you'd go to the build system and generate a new package or image containing the dependencies. Then why not run rosdep just on the build system?

issue.net on [poky-thud-20.0.1.tar.bz2](http://downloads.yoctoproject.org/releases/yocto/yocto-2.6.1/poky-thud-20.0.1.tar.bz2) is _"Poky (Yocto Project Reference Distro) 2.6.1 qemux86 /dev/ttyS0"_ . We may eventually decide on a way to detect OpenEmbedded but I'd prefer to use the override mechanism to avoid any misdetections and messing with other OSs for now.